### PR TITLE
Create new branches

### PR DESCRIPTION
**Pull request by mikegehard**
## Server Details

- Server: git
- Changes to: tools

## Motivation and Context
See https://github.com/modelcontextprotocol/servers/issues/147

## How Has This Been Tested?
I have tested creation of a branch from the current branch and from another branch in the Claude app.

## Breaking Changes
No

## Types of changes

- [ ] New MCP Server
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [X] Documentation update

## Checklist

- [X] I have read the [MCP Protocol Documentation](https://modelcontextprotocol.io)
- [ ] My server follows MCP security best practices
- [X] I have updated the server's README accordingly
- [X] I have tested this with an LLM client
- [X] My code follows the repository's style guidelines
- [ ] New and existing tests pass locally
- [ ] I have added appropriate error handling
- [ ] I have documented all environment variables and configuration options

### CODE REVIEW

File: src/git/README.md [hunk 1]
```diff
@@ -53,6 +53,13 @@ Please note that mcp-server-git is currently in early development. The functiona
      - `max_count` (number, optional): Maximum number of commits to show (default: 10)
    - Returns: Array of commit entries with hash, author, date, and message
 
+8. `git_branch`
```
Comment (jspahrsummers) [inline src/git/README.md:56]:
```suggestion
8. `git_create_branch`
```

File: src/git/src/mcp_server_git/server.py [hunk 2]
```diff
@@ -81,6 +87,15 @@ def git_log(repo: git.Repo, max_count: int = 10) -> list[str]:
...
     return log
 
+def git_create_branch(repo: git.Repo, branch_name: str, base_branch: str | None = None) -> str:
+    if base_branch:
+        base = repo.refs[base_branch]
+    else:
+        base = repo.active_branch
+
+    repo.create_head(branch_name, base)
+    return f"Created branch '{branch_name}' from {base.name}"
```
Comment (jspahrsummers) [inline src/git/src/mcp_server_git/server.py:97]:
```suggestion
    return f"Created branch '{branch_name}' from '{base.name}'"
```

**Review by jspahrsummers — DISMISSED** [review]
Great, thank you! Just a couple of nits to ensure consistency.

**Comment by mikegehard** [comment]
Probably best to squash this down into one commit on main.

**Review by jspahrsummers — APPROVED** [review]
That's okay, no need for squashing 🙂 

Thanks for fixing it up!

**Pull request merged**